packages/datadog-instrumentations/src/elasticsearch.js

```diff
@@ -22,6 +24,42 @@ addHook({ name: 'elasticsearch', file: 'src/lib/transport.js', versions: ['>=10'
   return Transport
 })
 
+addHook({ name: 'elasticsearch', file: 'src/lib/connection_pool.js', versions: ['>=10'] }, ConnectionPool => {
+  shimmer.wrap(ConnectionPool.prototype, 'select', createWrapSelect('elasticsearch'))
+  return ConnectionPool
+})
+
+function createWrapGetConnection (name) {
+  const connectCh = channel(`apm:${name}:query:connect`)
+  return function wrapRequest (request) {
+    return function () {
+      const connection = request.apply(this, arguments)
+      if (connectCh.hasSubscribers && connection?.url) {
+        connectCh.publish(connection.url)
+      }
+      return connection
+    }
+  }
+}
+
+function createWrapSelect () {
+  const connectCh = channel('apm:elasticsearch:query:connect')
+  return function wrapRequest (request) {
+    return function () {
+      if (arguments.length > 0) {
```

Comment:
Usually the callback is the final argument to a function. Does this function always have either exactly 1 or 0 argument?

If it always has exactly 1 or 0 arguments the check should probably be written like this:
```js
      if (arguments.length === 1) {
        const cb = arguments[0]
```

However, if there's a chance that there could ever be 2 or more arguments, we'd probably want to grab the last argument:

```js
 if (arguments.length > 0) {
        const cb = arguments[arguments.length - 1]
```

Or, if the function is a weird one that doesn't follow Node.js convention and the callback is always the first argument with possible other arguments, it might be worth throwing this in as a comment.

Reply:
yes is one arg function so it makes completely sense to check just for `=== 1` I applied the suggestion thanks